src/blocks/battery.rs

```diff
@@ -120,9 +119,52 @@ impl Block for Battery {
 
         let state = read_file(&format!("{}status", self.device_path))?;
 
+        let energy_now = if file_exists(&format!("{}energy_now", self.device_path)) {
+            read_file(&format!("{}energy_now", self.device_path))?
+                .parse::<u64>()
+                .block_error("battery", "failed to parse  energy_now")?
+        } else {
+            0
+        };
+
+
+
+        let energy_full = if file_exists(&format!("{}energy_full", self.device_path)) {
+             read_file(&format!("{}energy_full", self.device_path))?
+                .parse::<u64>()
+                .block_error("battery", "failed to parse  energy_full")?
+        } else {
+            0
+        };
+
+
+        let power_now = if file_exists(&format!("{}power_now", self.device_path)) {
+            read_file(&format!("{}power_now", self.device_path))?
+                .parse::<u64>()
+                .block_error("battery", "failed to parse current voltage")?
+        } else  {
+            0
+        };
+
+        let (hours, minutes) = if power_now > 0 && energy_now > 0 {
+            if state == "Discharging" {
+                let h = (energy_now as f64 / power_now as f64) as u64;
+                let m = (((energy_now  as f64 / power_now as f64) - h as f64)*60.0) as u64;
+                (h,m)
+            } else if state == "Charging" {
+                let h = ((energy_full as f64 - energy_now as f64) / power_now as f64) as u64;
+                let m = ((((energy_full as f64 - energy_now as f64)  / power_now as f64) - h as f64) * 60.0) as u64;
+                (h,m)
+            } else {
+                (0,0)
+            }
+        } else {
+            (0,0)
+        };
+
         // Don't need to display a percentage when the battery is full
         if current_percentage != 100 && state != "Full" {
-            self.output.set_text(format!("{}%", current_percentage));
+            self.output.set_text(format!("{}% {}:{:02}", current_percentage, hours, minutes));
```

Comment:
I removed the comma separator that can be seen in the screenshots between the percentage and the duration.